File: src/main/scala/com/vc/easy/L350.scala
package com.vc.easy

object L350 {
  def intersect(nums1: Array[Int], nums2: Array[Int]): Array[Int] = {
    import scala.collection.mutable
    val map = new mutable.HashMap[Int, Int]()
    val arr = new mutable.ArrayBuffer[Int]()
    nums1.foreach(x => {
      map.put(x, map.getOrElse(x, 0) + 1)
    })
    nums2.foreach(x => {
      if(map.contains(x) && map(x) > 0) {
        arr += x
        map.put(x, map(x) - 1)
      }
    })
    arr.toArray
  }
}

File: src/main/java/com/vc/easy/L350.java
package com.vc.easy;

import java.util.*;

class L350 {
    public int[] intersect(int[] nums1, int[] nums2) {
        HashMap<Integer, Integer> map = new HashMap<>();

        for(int i = 0; i < nums1.length; i++) {
            map.put(nums1[i], map.getOrDefault(nums1[i], 0) + 1);
        }

        List<Integer> list = new ArrayList<>();
        for(int i = 0; i < nums2.length; i++) {
            if(map.getOrDefault(nums2[i], 0) > 0) {
                list.add(nums2[i]);
                map.put(nums2[i], map.getOrDefault(nums2[i], 0) - 1);
            }
        }

        int[] res = new int[list.size()];
        for(int i = 0; i < list.size(); i++) res[i] = list.get(i);

        return res;
    }
}
